physical/postgresql.go

```diff
@@ -175,3 +199,143 @@ func (m *PostgreSQLBackend) List(prefix string) ([]string, error) {
 
 	return keys, nil
 }
+
+func (m *PostgreSQLBackend) HAEnabled() bool {
+	return true
+}
+
+// PostgreSQLLock implements the Lock interface for PostgreSQL
+type PostgreSQLLock struct {
+	client *sql.DB
+	table  string
+	leader chan struct{}
+
+	key     string
+	value   string
+	vaultID string
+}
+
+// LockWith initializes a Postgres backend lock
+func (m *PostgreSQLBackend) LockWith(key, value string) (Lock, error) {
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		return nil, err
+	}
+	return &PostgreSQLLock{
+		client:  m.client,
+		table:   m.lock_table,
+		leader:  make(chan struct{}),
+		key:     key,
+		value:   value,
+		vaultID: id,
+	}, nil
+}
+
+// Lock grabs a lock, or waits until it is available
+func (m *PostgreSQLLock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+	ticker := time.NewTicker(PostgreSQLLockRetryInterval)
+	defer ticker.Stop()
+	for {
+		_, err := m.client.ExecContext(
+			ctx,
+			"DELETE FROM "+m.table+" WHERE expiration < now()",
+		)
+		if err != nil {
+			return nil, err
+		}
+		lockTTL := strconv.Itoa(PostgreSQLLockTTL)
+		_, err = m.client.ExecContext(
+			ctx,
+			`INSERT INTO `+m.table+` (key, value, vault_id, expiration)
+				VALUES ($1, $2, $3, now() + interval '`+lockTTL+` seconds')`,
+			m.key,
+			m.value,
+			m.vaultID,
+		)
```

Comment:
Rely on the database driver to perform this logic now that lock expiration, `lockTTL`, is of type `time.Duration`.

```go
_, err := m.client.ExecContext(ctxt, lockSQL, m.key, m.value, m.vaultID, m.lockTTL.String())
```